qa/util/src/main/java/io/camunda/zeebe/qa/util/cluster/TestRestoreApp.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.qa.util.cluster;
+
+import io.camunda.zeebe.broker.system.configuration.BrokerCfg;
+import io.camunda.zeebe.restore.RestoreApp;
+import io.camunda.zeebe.shared.Profile;
+import java.util.function.Consumer;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+
+/** Represents an instance of the {@link RestoreApp} Spring application. */
+public final class TestRestoreApp extends TestSpringApplication<TestRestoreApp> {
+  private final BrokerCfg config;
+  private long backupId;
+
+  public TestRestoreApp() {
+    this(new BrokerCfg());
+  }
+
+  public TestRestoreApp(final BrokerCfg config) {
+    super(RestoreApp.class);
+    this.config = config;
+
+    //noinspection resource
+    withBean("config", config, BrokerCfg.class);
+  }
+
+  @Override
+  public TestRestoreApp self() {
+    return this;
+  }
+
+  @Override
+  protected String[] commandLineArgs() {
+    return new String[] {"--backupId=" + backupId};
```

Comment:
:wrench: As long as `withBackupId` is optional, this should probably return an empty array if `backupId` is null.

Reply:
Is it optional with `RestoreApp`? I guess...though we never test it. Let's do it :+1:

Reply:
Do you know what Spring will do? In `RestoreApp` it's also a long, so it can never be null. I assume not passing anything will make it 0?

Reply:
I suppose so :+1: It was more about that the user of `TestRestoreApp` can just not call `withBackupId`. Currently we'd treat that as "please restore backup 0". I thought just not providing the `--backupId` parameter would be better but I think you are right that Spring would still initialize the backup id to 0.

So nevermind, let's leave it like this

Reply:
Actually, I like the idea of not passing anything if not provided. Then it's closer to prod, meaning if we ever change how `RestoreApp` initializes `backupId`, this will match it :+1: